components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/logout/util/SAMLLogoutUtil.java

```diff
@@ -0,0 +1,305 @@
+import org.opensaml.saml.saml2.core.impl.StatusMessageBuilder;
+import org.opensaml.security.SecurityException;
+import org.wso2.carbon.identity.application.authenticator.samlsso.exception.SAMLSSOException;
+import org.wso2.carbon.identity.application.authenticator.samlsso.logout.context.SAMLMessageContext;
+import org.wso2.carbon.identity.application.authenticator.samlsso.logout.exception.SAMLLogoutException;
+import org.wso2.carbon.identity.application.authenticator.samlsso.logout.validators.LogoutReqSignatureValidator;
+import org.wso2.carbon.identity.application.authenticator.samlsso.manager.DefaultSAML2SSOManager;
+import org.wso2.carbon.identity.application.authenticator.samlsso.manager.X509CredentialImpl;
+import org.wso2.carbon.identity.application.authenticator.samlsso.util.SSOUtils;
+import org.wso2.carbon.identity.application.common.model.IdentityProvider;
+import org.wso2.carbon.identity.application.common.model.Property;
+import org.wso2.carbon.identity.base.IdentityException;
+import org.wso2.carbon.identity.saml.common.util.SAMLInitializer;
+
+import java.io.ByteArrayInputStream;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.wso2.carbon.identity.application.authenticator.samlsso.util.SSOConstants.StatusCodes.SUCCESS_CODE;
+import static org.wso2.carbon.identity.application.common.util.IdentityApplicationConstants.Authenticator.SAML2SSO.
+        SP_ENTITY_ID;
+import static org.wso2.carbon.identity.application.common.util.IdentityApplicationConstants.Authenticator.SAML2SSO.
+        SSO_URL;
+import static org.wso2.carbon.identity.application.common.util.IdentityApplicationConstants.Authenticator.SAML2SSO.
+        IS_AUTHN_RESP_SIGNED;
+import static org.wso2.carbon.identity.application.common.util.IdentityApplicationConstants.Authenticator.SAML2SSO.
+        IS_LOGOUT_REQ_SIGNED;
+import static org.wso2.carbon.identity.application.common.util.IdentityApplicationConstants.Authenticator.SAML2SSO.
+        INCLUDE_CERT;
+import static org.wso2.carbon.identity.application.common.util.IdentityApplicationConstants.Authenticator.SAML2SSO.
+        IS_SLO_REQUEST_ACCEPTED;
+import static org.wso2.carbon.identity.application.common.util.IdentityApplicationConstants.CERTIFICATE_TYPE;
+
+/**
+ * A Utility which provides functionality to handle federated idp initiated saml logout requests.
+ */
+public class SAMLLogoutUtil {
+
+    private static boolean bootStrapped = false;
+    private static final Log log = LogFactory.getLog(SAMLLogoutUtil.class);
+
+    private SAMLLogoutUtil() {
+
+    }
+
+    /**
+     * Bootstrap the OpenSAML3 library only if it is not bootstrapped.
+     *
+     */
+    public static void doBootstrap() {
+
+        // Initializing the OpenSAML library.
+        if (!bootStrapped) {
```

Comment:
Why do we need this check?

Reply:
To check whether OpenSAML library is already initialized or not.